### DISCUSSION TITLE
AnyOf implicitly allows additional properties on given models

### DISCUSSION THREAD

**Discussion by DKorosec** [TypedProject/ts-express-decorators#2825] [Q&A]
Hello 👋 

I've stumbled upon a problem where if I have a list of possible models (stored on a parent's model property), they implicitly allow additional properties. And only way to prevent it is to use `@AdditionalProperties(false)` on all of the data model classes in the list. 

Here is the example class
```typescript
class ModelA {
  @Required()
  a: string;
}

class ModelB {
  @Required()
  b: string;
}

class ExampleModel {
  @Required()
  id: string;

  @Required()
  @AnyOf(ModelA, ModelB)
  modelList: Array<ModelA | ModelB>;
}

@Controller("/example")
export class ExampleController {
  @Post("/")
  post(@BodyParams(ExampleModel) model: ExampleModel): ExampleModel {
    console.log(model);
    return model;
  }
}
```

where payload:
```json
{
  "id": "string",
  "additionalProperty": "this gets removed",
  "modelList": [
    {
      "a": "string",
      "additionalProperty": "this doesn't get removed"
    },
    {
      "b": "string"
    }
  ]
}
```

produces:
```json
{
  "id": "string",
  "modelList": [
    {
      "a": "string",
      "additionalProperty": "this doesn't get removed"
    },
    {
      "b": "string"
    }
  ]
}
```

Is this expected? And my question would be, is there a way to annotate ExampleModel's modelList so it doesn't allow additional properties implicitly (the same way they get removed in the parent model)?

**Comment by DKorosec** [comment]
I see this is closely related to https://github.com/orgs/tsedio/discussions/2781 and none of the proposed solutions (OneOf | Any | AnyOf) do work. But my question remains the same: is there a way to achieve this implicit additional properties removal instead of annotating classes with `AdditionalProperties(false)` ?

**Reply by Romakita** [reply]
Hi @DKorosec 

Sorry for the late answer. This option should resolve your problem:
https://beta.tsed.dev/docs/json-mapper.html#jsonmapper-additionalproperties

**Reply by Romakita** [reply]
But jsonMapper.additionaProperties is disabled by default.

**Reply by Romakita** [reply]
I'll check your issue ASAP
